packages/@aws-cdk/assertions/lib/match.ts

```diff
@@ -265,6 +273,42 @@ class ObjectMatch extends Matcher {
   }
 }
 
+class SerializedJson extends Matcher {
+  constructor(
+    public readonly name: string,
+    private readonly pattern: any,
+  ) {
+    super();
+  };
+
+  public test(actual: any): MatchResult {
+    const result = new MatchResult(actual);
+    if (getType(actual) !== 'string') {
+      result.push(this, [], `Expected JSON as a string but found ${getType(actual)}`);
+      return result;
+    }
+    let parsed;
+    try {
+      parsed = JSON.parse(actual);
+    } catch (err) {
+      if (err instanceof SyntaxError) {
+        result.push(this, [], 'Invalid JSON string');
```

Comment:
should this and the below include `actual` value?

also both should have tests

Reply:
Changed.

Added tests for one of them. The other is technically unreachable (or I'm not aware of how) but remains as a fail safe.

Reply:
Actually, I've removed this code block, since the `getType()` check above covers it.